datadog/dogstatsd/base.py

```diff
@@ -625,13 +625,37 @@ def _get_uds_socket(cls, socket_path, timeout):
 
     @classmethod
     def _get_udp_socket(cls, host, port, timeout):
-        sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+        sock = cls._create_udp_conn(host, port)
         sock.settimeout(timeout)
         cls._ensure_min_send_buffer_size(sock)
-        sock.connect((host, port))
-
         return sock
 
+    @classmethod
+    def _create_udp_conn(cls, host, port):
+        log.debug("Connecting to %s:%s", host, port)
+        addrinfo = socket.getaddrinfo(host, port, 0, socket.SOCK_DGRAM)
+        # Override gai.conf order for backwrads compatibility: prefer
+        # v4, so that a v4-only service on hosts with both addresses
+        # still works.
+        addrinfo.sort(key=lambda v: v[0] == socket.AF_INET, reverse=True)
+        lastaddr = len(addrinfo) - 1
+        for i, (af, ty, proto, _, addr) in enumerate(addrinfo):
+            sock = None
+            try:
+                sock = socket.socket(af, ty, proto)
+                sock.connect(addr)
```

Comment:
Since your change, are you sure it's OK to call the `settimeout` after the call to `connect`? Could be for UDP but might be worth preserving the old behavior: https://docs.python.org/3/library/socket.html#timeouts-and-the-connect-method

Reply:
I'm pretty sure socket timeout doesn't make any difference for udp sockets, but just for symmetry I updated the code to have the order same between udp and uds.